src/application/common/logger.ts

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable, unmanaged } from 'inversify';
+import { LoggerWatcher } from './logger-watcher';
+
+export namespace LogLevel {
+    export const FATAL = 60;
+    export const ERROR = 50;
+    export const WARN = 40;
+    export const INFO = 30;
+    export const DEBUG = 20;
+    export const TRACE = 10;
+}
+
+export const ILoggerServer = Symbol("ILoggerServer");
+
+export interface ILoggerServer {
+    setClient(client: ILoggerClient | undefined): void;
+    setLogLevel(id: number, logLevel: number): Promise<void>;
+    getLogLevel(id: number): Promise<number>;
+    log(id: number, logLevel: number, message: string, params: any[]): Promise<void>;
+    child(obj: Object): Promise<number>;
```

Comment:
Yes, prohibit primitives, I guess I was misslead by the bunyan d.ts with
 ``` child(obj: Object, simple?: boolean): Logger;```

I reading about it I think I'll use the object type, but I don't see why I would need to define that it could be extended ?

Should be like ```child(obj: object): Promise<number>;```
Since I intend to pass only { key: value} with key and value being primitives